### PR TITLE
Moved c,cpp, and python tests to tests folder. Add googletest as submodule

### PR DESCRIPTION
**Pull request by filipecosta90**
This PR moves c,cpp, and python tests to tests folder. Updates all scripts and automation accordingly.
It also removes googletest dependency from the src folder and adds it as a submodule.

### CODE REVIEW

**Comment by MeirShpilraien** [comment]
This a big change, I would like to wait with it until after we release 2.0. What do you think?

**Comment by rafie** [comment]
I think it's ok to wait until 2.0 is released.
I plan more changes (build/packaging/generic code), so we can start with this one after the release and then add more.
Note that we better cherry pick this PR into 1.6/1.8.

File: Makefile [hunk 1]
```diff
@@ -170,18 +169,15 @@ endif
...
 ifeq ($(TEST),)
 	set -e ;\
-	cd src/tests ;\
-	$(GDB_CMD) $(abspath $(BINROOT)/src/cpptests/rstest)
+	find $(abspath $(BINROOT)/tests/cpptests) -name "test_*" -type f -executable -exec ${GDB_CMD} {} \;
 else
 	set -e ;\
-	cd src/tests ;\
-	$(GDB_CMD) $(abspath $(BINROOT)/src/cpptests/rstest) --gtest_filter=$(TEST)
+	$(GDB_CMD) $(abspath $(BINROOT)/tests/cpptests/$(TEST))
```
Comment (rafie) [inline Makefile:180]: Can we keep the `--gtest_filter=$(TEST)` argument?

File: Makefile [hunk 2]
```diff
@@ -170,18 +169,15 @@ endif
...
+	find $(abspath $(BINROOT)/tests/ctests) -name "test_*" -type f -executable -exec ${GDB_CMD} {} \;
 
 cpp_tests:
 ifeq ($(TEST),)
 	set -e ;\
-	cd src/tests ;\
-	$(GDB_CMD) $(abspath $(BINROOT)/src/cpptests/rstest)
+	find $(abspath $(BINROOT)/tests/cpptests) -name "test_*" -type f -executable -exec ${GDB_CMD} {} \;
 else
 	set -e ;\
```
Comment (rafie) [inline Makefile:179]: we can remove `set -e`

File: Makefile [hunk 3]
```diff
@@ -170,18 +169,15 @@ endif
 
 c_tests:
 	set -e ;\
```
Comment (rafie) [inline Makefile:171]: we can remove `set -e`

File: Makefile [hunk 4]
```diff
@@ -157,9 +156,9 @@ pytest:
 		exit 1 ;\
 	fi
 ifneq ($(TEST),)
-	@cd src/pytest; PYDEBUG=1 python -m RLTest --test $(TEST) $(RLTEST_GDB) -s --module $(abspath $(TARGET))
+	@cd tests/pytests; PYDEBUG=1 python -m RLTest --test $(TEST) $(RLTEST_GDB) -s --module $(abspath $(TARGET))
```
Comment (rafie) [inline Makefile:159]: need to add `set -e` at the top of the `pytest` rule.

File: Makefile (hunk 1 above)
Comment (filipecosta90) [reply Makefile]: :+1: solved in the last commit.

**Review by rafie — APPROVED** [review]

**Review by MeirShpilraien — APPROVED** [review]

**Pull request merged**